www/js/metrics/CarbonTextCard.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useMemo } from 'react';
+import { View } from 'react-native';
+import { Card, Text, useTheme} from 'react-native-paper';
+import { MetricsData } from './metricsTypes';
+import { cardStyles } from './MetricsTab';
+import { useTranslation } from 'react-i18next';
+import { formatDateRangeOfDays, parseDataFromMetrics, generateSummaryFromData, calculatePercentChange, segmentDaysByWeeks } from './metricsHelper';
+import { getAngularService } from '../angular-react-helper';
+
+type Props = { userMetrics: MetricsData, aggMetrics: MetricsData }
+const CarbonTextCard = ({ userMetrics, aggMetrics }: Props) => {
+
+  const { colors } = useTheme();
+  const { t } = useTranslation();
+  const FootprintHelper = getAngularService("FootprintHelper");
+
+  const userText = useMemo(() => {
+    if(userMetrics?.distance?.length > 0) {
+        //separate data into weeks
+        const [thisWeekDistance, lastWeekDistance] = segmentDaysByWeeks(userMetrics?.distance, 2);
+
+        //formatted data from last week
+        let userLastWeekModeMap = {};
+        let userLastWeekSummaryMap = {};
+        if(lastWeekDistance && lastWeekDistance?.length >= 7) {
+            userLastWeekModeMap = parseDataFromMetrics(lastWeekDistance, 'user');
+            userLastWeekSummaryMap = generateSummaryFromData(userLastWeekModeMap, 'distance');
+        }
+        
+        //formatted distance data from this week
+        let userThisWeekModeMap = parseDataFromMetrics(thisWeekDistance, 'user');
+        let userThisWeekSummaryMap = generateSummaryFromData(userThisWeekModeMap, 'distance');
+        let worstDistance = userThisWeekSummaryMap.reduce((prevDistance, currModeSummary) => prevDistance + currModeSummary.values, 0);
+        
+        //setting up data to be displayed
+        let textList = [];
+
+        //calculate low-high and format range for prev week, if exists
+        if(userLastWeekSummaryMap[0]) {
+          let userPrevWeek = {
+              low: FootprintHelper.getFootprintForMetrics(userLastWeekSummaryMap, 0), 
+              high: FootprintHelper.getFootprintForMetrics(userLastWeekSummaryMap, FootprintHelper.getHighestFootprint())
+          };
+          const label = `${t('main-metrics.prev-week')} (${formatDateRangeOfDays(lastWeekDistance)})`;
+          if (userPrevWeek.low == userPrevWeek.high)
+            textList.push({label: label, value: Math.round(userPrevWeek.low)});
+          else
+            textList.push({label: label + '²', value: `${Math.round(userPrevWeek.low)} - ${Math.round(userPrevWeek.high)}`});
+        }
```

Comment:
seems like all this computation can also become common.

Reply:
TODO cleanup